workerbee/postgresql.py

```diff
@@ -28,257 +34,328 @@ def check_name(table_name):
 
 
-def set_job_as_claimed(db_handle, table_id, job_id):
-    db_handle.run(SET_ROW_CLAIMED_BY_ID_QUERY.format(table_id),
-                  parameters={'job_id': job_id})
+def set_job_as_complete(db_handle, tbl_name, job_id, duration,
+                        output_data=None):
+    if output_data is not None:
+        output_data = postgres_jsonify(output_data)
+    db_handle.run(SET_ROW_COMPLETED_BY_ID_QUERY.format(tbl_name=tbl_name),
+                  parameters={'id': job_id, 'job_duration': duration,
+                              'output_data': output_data})
 
 
-def get_total_job_count(db_handle, table_id):
-    return db_handle.one(TOTAL_ROWS_QUERY.format(table_id))
+def set_job_as_claimed(db_handle, tbl_name, job_id):
+    db_handle.run(SET_ROW_CLAIMED_BY_ID_QUERY.format(tbl_name=tbl_name),
+                  parameters={'id': job_id})
 
 
-def get_completed_job_count(db_handle, table_id):
-    return db_handle.one(COMPLETED_ROWS_QUERY.format(table_id))
+def update_job_n_failed_attempts(db_handle, tbl_name, job_id):
+    db_handle.run(UPDATE_ROW_N_FAILED_ATTEMPTS_BY_ID_QUERY.format(tbl_name=tbl_name),
+                  parameters={'id': job_id})
+
+
+def get_total_job_count(db_handle, tbl_name):
+    return db_handle.one(TOTAL_ROWS_QUERY.format(tbl_name=tbl_name))
+
+
+def get_completed_job_count(db_handle, tbl_name):
+    return db_handle.one(COMPLETED_ROWS_QUERY.format(tbl_name=tbl_name))
 
 ################################################################################
 
 
-def postgres_experiment(experiment_id, job_function,
-                        job_ids=None, job_data=None,
-                        host=None, port=None, user=None, database=None,
-                        verbose=True):
+def add_job(experiment_id, input_data, unique_id=None, db_connection_info=None,
+            logger_name=DEFAULT_LOGGER, cursor=None):
+    query_str = INSERT_JOB_QUERY.format(tbl_name=experiment_id)
+    params = {'parameters': {'unique_id': unique_id,
+                             'input_data': postgres_jsonify(input_data)}}
+    if cursor is None:
+        db_handle = get_db_handle(db_info=db_connection_info,
+                                  logger_name=logger_name)
+
+        if not table_exists(db_handle, experiment_id):
+            raise ValueError("Table does not exist for experiment '{}'".format(
+                experiment_id))
+
+        db_handle.run(query_str, **params)
+    else:
+        cursor.run(query_str, **params)
+
+
+def add_jobs(experiment_id, input_datas, unique_ids=None,
+             db_connection_info=None, logger_name=DEFAULT_LOGGER, cursor=None):
+    logger = logging.getLogger(logger_name)
+    if cursor is None:
+        db_handle = get_db_handle(db_info=db_connection_info,
+                                  logger_name=logger_name)
+
+        if not table_exists(db_handle, experiment_id):
+            raise ValueError("Table does not exist for experiment '{}'".format(
+                experiment_id))
```

Comment:
Thinking a little further, maybe the best option is 1 (explicit), but with a keyword arg `autocreate=False`.

You could then do:
```py
from workerbee.postgres import add_jobs
add_jobs('my_new_exp', [{'path': 'thepath'}], autocreate=True)
```
In the absence of that kwarg I could imagine this being a common pattern:
```py
from workerbee.postgres import add_jobs, create_experiment
try:
    create_experiment('my_exp')
catch ASuitableException:
    pass
add_jobs('my_new_exp', [{'path': 'thepath'}])
```
(at least for the way I currently use workerbee). At least this way the user is explicitly opting in to the more loosey goosey behaviour (like `overwrite=True`)